src/_imaging.c

```diff
@@ -3707,16 +3704,39 @@ _getattr_bands(ImagingObject *self, void *closure) {
 
 static PyObject *
 _getattr_id(ImagingObject *self, void *closure) {
+    if (PyErr_WarnEx(
+            PyExc_DeprecationWarning,
+            "id property is deprecated and will be removed in Pillow 12.0",
```

Comment:
```suggestion
            "id property is deprecated and will be removed in Pillow 12 (2025-10-15)",
```
For consistency with _deprecate.py.